pkg/ovs/openflow/ofctrl_packetout.go

```diff
@@ -403,17 +403,24 @@ func (b *ofPacketOutBuilder) generatePseudoHeader(length uint16) []byte {
 }
 
 func checksum(data []byte) uint16 {
-	var sum uint32
-	var index int
-	length := len(data)
-	for length > 1 {
-		sum += uint32(data[index])<<8 + uint32(data[index+1])
-		index += 2
-		length -= 2
+	sum := uint32(0)
+	for ; len(data) >= 2; data = data[2:] {
+		sum += uint32(data[0])<<8 | uint32(data[1])
 	}
-	if length > 0 {
-		sum += uint32(data[index])
+	if len(data) > 0 {
+		sum += uint32(data[0]) << 8
 	}
-	sum += (sum >> 16)
-	return uint16(^sum)
+	for sum > 0xffff {
+		sum = (sum >> 16) + (sum & 0xffff)
```

Comment:
Do we need to do this operation twice? I think this operation may still generate a carry to higher 16 bits?